test/distributed/org/apache/cassandra/distributed/mock/nodetool/InternalNodeProbe.java

```diff
@@ -61,7 +79,26 @@ protected void connect()
         mbeanServerConn = null;
         jmxc = null;
 
-        ssProxy = StorageService.instance;
+        if (withNotifications)
+        {
+            ssProxy = StorageService.instance;
+        }
+        else
+        {
+            // replace the notification apis with a no-op method
+            StorageServiceMBean mock = Mockito.spy(StorageService.instance);
+            Mockito.doNothing().when(mock).addNotificationListener(Mockito.any(), Mockito.any(), Mockito.any());
+            try
+            {
+                Mockito.doNothing().when(mock).removeNotificationListener(Mockito.any(), Mockito.any(), Mockito.any());
+                Mockito.doNothing().when(mock).removeNotificationListener(Mockito.any());
+            }
+            catch (ListenerNotFoundException e)
+            {
+                throw new AssertionError(e);
+            }
+            ssProxy = mock;
```

Comment:
I don't understand what are the cases that a no-op storage-service is needed..

Reply:
Very very good question!

This only happens when `withNotifications == false`.  The thing about JMX notifications are they are lossy.  Given the lossy nature we need the ability to test that we do the right thing when notifications are not seen on the caller.  So we disable notifications by creating a "spy" on StorageService and making `removeNotificationListener` do nothing.